lib/upload.js

```diff
@@ -615,6 +615,11 @@ class BaseUpload {
         if (status === 423) {
           this._emitHttpError(req, res, 'tus: upload is currently locked; retry later')
           return
+        } else if (inStatusCategory(status, 500)) {
+          // Run retry logic if the server has an error, e.g. 502 Bad Gateway when
+          // proxied server is temporarily down. See issue #579.
```

Comment:
Please mention that we do not want to clear `this._url` and create a new upload here.

Reply:
Done.